classes/Langchecker/DotLangParser.php

```diff
@@ -122,57 +133,88 @@ function ($tag) {
                          */
                         $dotlang_data['strings'][$reference] = $reference;
                         continue;
-                    } else {
-                        // Store the translation
-                        $dotlang_data['strings'][$reference] = $translation;
                     }
+
+                    // Store the translation
+                    $dotlang_data['strings'][$reference] = $translation;
+
                     $i++;
                 }
             }
 
             // Second pass: extract only metadata (comments, tag bindings) for reference locale.
             if ($reference_locale) {
-                for ($i = 0, $lines = count($file_content); $i < $lines; $i++) {
-                    $current_line = $file_content[$i];
-                    if (Utils::startsWith($current_line, ';')) {
-                        // I have a reference string
-                        $reference = Utils::leftStrip($current_line, ';');
-                        $j = $i - 1;
-                        while ($j > 0) {
-                            // Stop if I find a line not starting with #
-                            if (! Utils::startsWith($file_content[$j], '#')) {
-                                break;
-                            }
-                            // Comments
-                            if (Utils::startsWith($file_content[$j], '#') &&
-                                ! Utils::startsWith($file_content[$j], '##')) {
-                                $dotlang_data['comments'][$reference][] = Utils::leftStrip($file_content[$j], '#');
-                            } else {
-                                // Tag bindings
-                                if (Utils::startsWith($file_content[$j], '## TAG:')) {
-                                    $dotlang_data['tag_bindings'][$reference] = Utils::leftStrip($file_content[$j], '## TAG:');
-                                }
-                                // Length limits
-                                if (Utils::startsWith($file_content[$j], '## MAX_LENGTH:')) {
-                                    $dotlang_data['max_lengths'][$reference] = intval(Utils::leftStrip($file_content[$j], '## MAX_LENGTH:'));
-                                }
-                            }
-                            $j--;
+                $dotlang_data = array_merge($dotlang_data, self::extractReferenceMetaData($file_content));
+                self::$parsed_files[$path] = $dotlang_data;
+            }
+        }
+
+        return $dotlang_data;
+    }
+
+    /**
+     * Provide all meta tags in format “## METATAG:”,
+     * but not file tags (“## TAGNAME ##”)
+     *
```

Comment:
Leave only the first line of the comment, the second one is not relevant for the function.